node/pkg/solana/client.go

```diff
@@ -386,7 +387,7 @@ func (s *SolanaWatcher) processInstruction(ctx context.Context, logger *zap.Logg
 		return false, nil
 	}
 
-	if inst.Data[0] != postMessageInstructionID {
```

Comment:
Unrelated but do we need to check that len(inst.Data) > 0 first? Looks like this is coming from untrusted input.

Reply:
Added this check